platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -375,28 +389,37 @@ private String getEntityId() {
     return String.format("https://%s:%s%s/saml", hostname, port, rootContext);
   }
 
-  private void logout() {
-    HttpSession session = sessionFactory.getOrCreateSession(request);
+  private SecurityAssertion getIdpSecurityAssertion() {
 
-    SecurityTokenHolder tokenHolder =
-        ((SecurityTokenHolder) session.getAttribute(SecurityConstants.SAML_ASSERTION));
+    return new SecurityAssertionImpl(getTokenHolder().getSecurityToken(IDP_REALM_NAME));
+  }
 
-    SecurityAssertion securityAssertion =
-        new SecurityAssertionImpl(tokenHolder.getSecurityToken("idp"));
+  private HttpSession getOrCreateSession() {
+    return sessionFactory.getOrCreateSession(request);
+  }
 
-    boolean hasSecurityAuditRole =
-        Arrays.stream(System.getProperty("security.audit.roles").split(","))
-            .filter(role -> securityAssertion.getPrincipals().contains(new RolePrincipal(role)))
-            .findFirst()
-            .isPresent();
+  private void logout() {
+    logSecurityAuditRole();
+    getTokenHolder().remove(IDP_REALM_NAME);
+  }
 
-    if (hasSecurityAuditRole) {
+  private void logSecurityAuditRole() {
+    if (shouldAuditSubject()) {
       SecurityLogger.audit(
           "Subject with admin privileges has logged out: {}",
-          securityAssertion.getPrincipal().getName());
+          getIdpSecurityAssertion().getPrincipal().getName());
     }
+  }
 
-    tokenHolder.remove("idp");
+  private boolean shouldAuditSubject() {
+    return Arrays.stream(System.getProperty("security.audit.roles").split(","))
+        .anyMatch(
+            role -> getIdpSecurityAssertion().getPrincipals().contains(new RolePrincipal(role)));
+  }
+
+  private SecurityTokenHolder getTokenHolder() {
```

Comment:
Looked at the code and looks like getAttribute can return null, so we should annotate this method with `@Nullable`

Reply:
Right.